### PR TITLE
ci: Upload source maps to s3

### LINKED ISSUE #18063: [Task]: Upload client source maps to S3
### Is there an existing issue for this?

- [X] I have searched the existing issues

### SubTasks

Upload source maps to s3 for use while debugging performance runs, and deployed applications.

### PR DESCRIPTION
**Pull request by SatishGandham**
## Description

- Upload source maps to a private s3 bucket
- Delete zipped version of source maps

Fixes #18063

## Type of change
- Non breaking enhancement to CI

### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### CODE REVIEW

File: app/client/upload-source-maps.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+const BUCKET_NAME = "performance-infra-artifacts-appsmith";
+// Create S3 service object
+const s3 = new AWS.S3({
+  accessKeyId: key,
+  secretAccessKey: secret,
+});
+
+uploadFiles = async () => {
+  if (!process.env.CI) {
```
Comment (mohanarpit) [inline app/client/upload-source-maps.js:17]: Any reason we put the if-check here instead of `build.sh`?

**Review by mohanarpit — DISMISSED** [review]

File: app/client/upload-source-maps.js (hunk 1 above)
Comment (SatishGandham) [reply app/client/upload-source-maps.js:17]: I'm not comfortable writing bash.

**Pull request closed without merging**